nscplugin/src/main/scala/scala/scalanative/nscplugin/NirGenFile.scala

```diff
@@ -29,10 +27,8 @@ trait NirGenFile { self: NirGenPhase =>
     Paths.get(file.file.getAbsolutePath)
   }
 
-  def genIRFile(path: Path, defns: Seq[nir.Defn]): Unit =
-    withScratchBuffer { buffer =>
-      serializeBinary(defns, buffer)
-      buffer.flip
-      VirtualDirectory.local(path.getParent).write(path, buffer)
-    }
+  def genIRFile(path: Path, defns: Seq[nir.Defn]): Unit = {
+    val outStream = new FileOutputStream(path.toFile)
+    serializeBinary(defns, outStream)
+  }
```

Comment:
At another hand this code has at least one issue: you never close `outStream` and it is possibly that sbt gets Too Many Open Files.

Trivial changes fixes the code:
```
    val outStream = new FileOutputStream(path.toFile)
    try {
      serializeBinary(defns, outStream)
    } finally {
      outStream.close()
    }
```

Reply:
Indeed, we definitely need to `try ... finally outStream.close()` here.